SHARE-124 - DB migration and implementation of UserAPI

bonbuo opened the pull request:
added DB migration for ranking_score and implemented getUsers function

---

### Your checklist for this pull request

Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [ ] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [ ] Choose the proper base branch (_develop_)
- [ ] Confirm that the changes follow the project’s coding guidelines
- [ ] Verify that the changes generate no warnings and errors
- [ ] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Verify that all tests are passing (CI), if not please state the test cases in the [section](#Tests) below
- [ ] Perform a self-review of the changes
- [ ] Stick to the project’s git workflow (rebase and squash your commits)
- [ ] Verify that your changes do not have any conflicts with the base branch
- [ ] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [ ] Post a message in the _#catroweb_ [Slack channel](https://catrobat.slack.com) and ask for a code reviewer
- [ ] Check that your pull request has been successfully deployed to https://web-test-1.catrob.at/

### Additional Description

`TODO: Add additional information that is not in your commit-message here`

### Tests - additional information

`TODO: add additional information about testruns here`

dependabot[bot] committed:
Build(deps-dev): Bump eslint from 8.55.0 to 8.56.0

Bumps [eslint](https://github.com/eslint/eslint) from 8.55.0 to 8.56.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.55.0...v8.56.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

bonbuo committed:
SHARE-124 - DB migration and implementation of UserAPI

added DB migration for ranking_score and implemented getUsers function

bonbuo committed:
Merge branch 'develop' into SHARE-124

bonbuo committed:
SHARE-124 Symfony Command to compute ELO

implemented the Symfony command to recompute the ELO for all users

bonbuo committed:
Merge branch 'SHARE-124' of github.com:bonbuo/Catroweb into SHARE-124

bonbuo committed:
SHARE-124 User Ranking

added test for the symfony command which updates the rankingScore of all users

bonbuo committed:
SHARE-124 User Ranking

added test for the symfony command which updates the rankingScore of all users

dmetzner commented:
@bonbuo please rebase - I have added your version + the stub to develop already; Otherwise you would merge conflicts in the composer files all the time. That's the reason why it's really helpful to create multiple smaller pull requests. Something that could also be easily done for the DB migration :)

dmetzner commented on migrations/2023/Version20231220122134.php:
```diff
@@ -0,0 +1,41 @@
+        $this->addSql('ALTER TABLE studio_activity CHANGE type type ENUM(\'comment\', \'project\', \'user\')');
+        $this->addSql('ALTER TABLE studio_user CHANGE role role ENUM(\'admin\', \'member\'), CHANGE status status ENUM(\'active\', \'banned\', \'pending_request\')');
+        $this->addSql('ALTER TABLE user_like_similarity_relation CHANGE similarity similarity NUMERIC(4, 3) DEFAULT \'0\' NOT NULL');
+        $this->addSql('ALTER TABLE user_remix_similarity_relation CHANGE similarity similarity NUMERIC(4, 3) DEFAULT \'0\' NOT NULL');
```
those lines are noise and can be removed

dmetzner commented on migrations/2023/Version20231220122134.php:
```diff
@@ -0,0 +1,41 @@
+    {
+        // this down() migration is auto-generated, please modify it to your needs
+        $this->addSql('ALTER TABLE user_like_similarity_relation CHANGE similarity similarity NUMERIC(4, 3) DEFAULT \'0.000\' NOT NULL');
+        $this->addSql('ALTER TABLE user_remix_similarity_relation CHANGE similarity similarity NUMERIC(4, 3) DEFAULT \'0.000\' NOT NULL');
```
those lines are noise and can be removed

dmetzner commented on migrations/2023/Version20231220122134.php:
```diff
@@ -0,0 +1,41 @@
+        $this->addSql('DROP INDEX UNIQ_957A64793F5D60E0 ON fos_user');
+        $this->addSql('ALTER TABLE fos_user DROP ranking_score');
+        $this->addSql('ALTER TABLE studio_user CHANGE role role VARCHAR(255) DEFAULT NULL, CHANGE status status VARCHAR(255) DEFAULT NULL');
+        $this->addSql('ALTER TABLE studio_activity CHANGE type type VARCHAR(255) DEFAULT NULL');
```
those lines are noise and can be removed

dmetzner commented on src/Api/Services/User/UserResponseManager.php:
```diff
@@ -28,7 +28,7 @@ public function createBasicUserDataResponse(User $user, string $attributes = nul
       $attributes_list = ['id', 'username'];
     } elseif ('ALL' === $attributes) {
-      $attributes_list = ['id', 'username', 'picture', 'about', 'currentlyWorkingOn', 'projects', 'followers', 'following'];
+      $attributes_list = ['id', 'username', 'picture', 'about', 'currentlyWorkingOn', 'projects', 'followers', 'following', 'ranking_score'];
```
we should try to keep a consistent naming convention;

currentlyWorkingOn vs ranking_score

dmetzner replied:
Maybe you could change `currentlyWorkingOn`; pretty sure no one is using it already, but please double check

bonbuo replied:
it seems there are some translations and already some usages for currentlyWorkingOn

dmetzner replied:
yes, but only on our side, it is not a feature used by the apps yet :)
Can you please fix this in a new merge request?

dmetzner commented on src/System/Commands/DBUpdater/UpdateUserRankingCommand.php:
```diff
@@ -0,0 +1,68 @@
+
+        $users = $this->userRepository->findAll();
+
+        foreach ($users as $user) {
```
consider doing the computation in batches, we have a lot of users.

bonbuo committed:
Build(deps-dev): Bump vimeo/psalm from 5.17.0 to 5.18.0

Bumps [vimeo/psalm](https://github.com/vimeo/psalm) from 5.17.0 to 5.18.0.
- [Release notes](https://github.com/vimeo/psalm/releases)
- [Commits](https://github.com/vimeo/psalm/compare/5.17.0...5.18.0)

---
updated-dependencies:
- dependency-name: vimeo/psalm
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

bonbuo committed:
Merge branch 'SHARE-124' of github.com:bonbuo/Catroweb into SHARE-124

bonbuo committed:
SHARE-124 removing noise lines from Version files

removed noise and refactoring

bonbuo committed:
SHARE-124 removed further noise lines

removed noise lines from the DB-Migration Version file

dmetzner committed:
Merge branch 'develop' into SHARE-124
